Subtracting a brush that encloses the subtrahend has no effect.

There's currently also some issue that subtract doesn't produce correct result if the first brush is completely enclosed in the second. The second brush just disappears and first brush stays untouched.

Both brushes should disappear entirely.

I tried to look at it yesterday, but I couldn't get git master to compile. Tag v2.0.6 compiles fine with these spells, but there's some changes to that subtraction code after that.

```
export CXX=/usr/bin/clang++-6.0
export CC=/usr/bin/clang-6.0
cmake .. -DCMAKE_BUILD_TYPE=Debug -DwxWidgets_PREFIX=/usr -DCMAKE_CXX_FLAGS=-std=c++17
cmake --build . --target TrenchBroom -- -j4
```
And it fails with
```
[ 30%] Building CXX object CMakeFiles/common.dir/common/src/View/EntityAttributeSelectedCommand.cpp.o
/home/mankeli/src/TrenchBroom/common/src/View/ScaleObjectsToolController.cpp:96:20: error: chosen constructor is explicit in copy-initialization
            return {restricter, snapper, initialPoint};
                   ^~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:479:19: note: explicit constructor declared here
        constexpr tuple(_UElements&&... __elements)
                  ^
```

during configure I also get messages like:
```
CMake Error in CMakeLists.txt:
  Target "TrenchBroom-Test" requires the language dialect "CXX17" (with
  compiler extensions), but CMake does not know the compile flags to use to
  enable it.
```

(ubuntu, clang 6.0)

--- a/common/src/View/MapDocument.cpp
+++ b/common/src/View/MapDocument.cpp
@@ -1090,10 +1090,11 @@ namespace TrenchBroom {
             
             for (Model::Brush* minuend : minuends) {
                 const Model::BrushList result = minuend->subtract(*m_world, m_worldBounds, currentTextureName(), subtrahend);
+
                 if (!result.empty()) {
                     VectorUtils::append(toAdd[minuend->parent()], result);
-                    toRemove.push_back(minuend);
                 }
+                toRemove.push_back(minuend);
             }
             
             Transaction transaction(this, "CSG Subtract");

--- a/test/src/Model/BrushTest.cpp
+++ b/test/src/Model/BrushTest.cpp
@@ -2643,6 +2643,25 @@ namespace TrenchBroom {
 
             VectorUtils::deleteAll(result);
         }
+
+        TEST(BrushTest, subtractEnclosed) {
+            const vm::bbox3 worldBounds(4096.0);
+            World world(MapFormat::Standard, nullptr, worldBounds);
+            
+            const vm::bbox3 brush1Bounds(vm::vec3::fill(-8.0), vm::vec3::fill(+8.0));
+            const vm::bbox3 brush2Bounds(vm::vec3::fill(-9.0), vm::vec3::fill(+9.0));
+            ASSERT_TRUE(brush1Bounds.intersects(brush2Bounds));
+            
+            BrushBuilder builder(&world, worldBounds);
+            Brush* brush1 = builder.createCuboid(brush1Bounds, "texture");
+            Brush* brush2 = builder.createCuboid(brush2Bounds, "texture");
+            
+            BrushList result = brush1->subtract(world, worldBounds, "texture", brush2);
+            ASSERT_EQ(0u, result.size());
+            
+            VectorUtils::deleteAll(result);
+        }
+
         
         TEST(BrushTest, subtractTruncatedCones) {
             // https://github.com/kduske/TrenchBroom/issues/1469

